client/src/components/BoxDisplay.js

```diff
@@ -29,22 +27,45 @@ function BoxDisplay({ studentId, studentName }) {
         }
         console.log(data);
         setStudentDetail(data);
-        // setIsClicked(!isClicked);
       });
   };
+
   useEffect(fetchDetails, [studentId]);
+  const abilityLength = [];
+  console.log(abilityLength);
 
-  const details = (skill) => {
+  const displayDetails = (skill) => {
     return studentDetail
       .filter((lo) => lo.skill === skill)
       .map(({ description, ability }) => {
         return (
           <p>
-            - {description}:<button>score :{ability}</button>
+            - {description}: <button> score :{ability} </button>
           </p>
         );
       });
   };
+
+  const getAverageAbility = (skill) => {
+    const filteredResults = studentDetail.filter((lo) => lo.skill === skill);
```

Comment:
I wonder if you also want to filter out the learning objectives that don't have any answer? At the moment these will be affecting the averages. But maybe this is a question for you to ask the product owner?